src/protocol/http.rs

```diff
@@ -0,0 +1,150 @@
+use crate::extend::ExtendableThing;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, skip_serializing_none};
+
+#[derive(Debug, Clone, Deserialize, Serialize, Copy, PartialEq, Eq, Hash)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum Method {
+    Get,
+    Put,
+    Post,
+    Delete,
+    Patch,
+}
+
+#[serde_as]
+#[skip_serializing_none]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash, Default)]
+struct MessageHeader {
```

Comment:
_Maybe_ we are missing `htv:hdrName` and `htv:headerElements`, but I am not sure.

The [vocabulary terms table](https://w3c.github.io/wot-binding-templates/bindings/protocols/http/index.html#http-vocabulary-terms) includes a link to [HTTP-in-RDF10](https://www.w3.org/TR/HTTP-in-RDF10/#MessageHeaderClass) (just to make things harder), which includes these two additional params.

I totally don't understand which type they are expected to be.

Reply:
That part is in flux, I'd keep what we have and wait for @egekorkan to update the specification.